frontend/util/multi-client-select.component.tsx

```diff
@@ -0,0 +1,77 @@
+import { maybe } from "kremling";
+import React from "react";
+import SingleClientSearchInput from "../client-search/single-client/single-client-search-input.component";
+import CUProgramInputs from "../programs-and-services/cu-program-inputs.component";
+import { SingleClient } from "../view-edit-client/view-client.component";
+import CloseIconButton from "./close-icon-button.component";
+
+const MultiClientSelect = React.forwardRef<
+  MultiClientSelectRef,
+  MultiClientSelectProps
+>((props, ref) => {
+  const [clients, setClients] = React.useState(props.initialClients);
+  const [showEmptyRow, setShowEmptyRow] = React.useState(
+    props.initialClients.length === 0
+  );
+  React.useImperativeHandle(ref, () => ({
+    getClients() {
+      return clients;
+    },
+  }));
+
+  const removeClientSelect = () => {
+    let newArr = [...clients];
+    newArr.pop();
+    setClients(newArr);
+  };
+
+  return (
+    <div
+      className={maybe(
+        props.containerClass || "",
+        Boolean(props.containerClass)
+      )}
+    >
+      {clients.map((client, idx) => (
+        <div
+          className={maybe(props.clientClass || "", Boolean(props.clientClass))}
+          key={idx}
```

Comment:
It's better to use `client.id` here as the `key` instead of the array index, since the array index could theoretically change but the client id wouldn't. The `key` prop is allow about handling the case where the order of arrays change, so passing in the index is not really helpful (even though it makes the pesky warning go away 😄)

```suggestion
          key={client.id}
```